pkg/apis/kubeless/v1beta1/kafka_trigger.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright (c) 2016-2017 Bitnami
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KafkaTrigger object
+type KafkaTrigger struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              KafkaTriggerSpec `json:"spec"`
+}
+
+// KafkaTriggerSpec contains func specification
+type KafkaTriggerSpec struct {
+	Topic            string               `json:"topic"` // Trigger topic (for PubSub type)
+	FunctionSelector metav1.LabelSelector `json:"functionSelector"`
```

Comment:
For `CronJob` and `Http` triggers they are naturally can be associated with a single function. So it did not make sense to use label selector. For e.g a cron job, running multiple functions at periodic interval is confusing. For http trigger it is 1:1 any way.

Reply:
... so should KafkaTrigger also drop the label selector?  Surely each Trigger declaration is also a single (topic, function) ?

Reply:
Some of the users have suggested to use label selector in case of Kafka, so that we have a way to achieve n:m association between function and Kafka event source.